### PR TITLE
Expand comment explaining update_one_slot

### PR DESCRIPTION
**Pull request by jdemeyer**
This is just changing comments in the code, so `skip issue` and `skip news`.

CC @vstinner

### CODE REVIEW

File: Objects/typeobject.c [hunk 1]
```diff
@@ -7203,12 +7204,61 @@ resolve_slotdups(PyTypeObject *type, PyObject *name)
...
+ * resulting slot is applied. If the results disagree, then a wrapper for
+ * the special methods is installed. This is always safe, but less efficient
+ * because it uses method lookup instead of direct C calls.
+ *
+ * There are some further special cases for specific slots, like supporting
+ * __hash__ = None for tp_hash and special code for tp_new.
+ *
+ * When done, return a pointer to the next slotdef with a different offset,
+ * because that's convenient for fixup_slot_dispatchers(). This function never
+ * sets an exception: if an internal error happens (unlikely), it's ignored. */
```
Comment (epicfaace) [inline Objects/typeobject.c:7261]: Is it true that this function never sets an exception? #14836 seems to have an example in which it does set an exception.

Comment (jdemeyer) [reply Objects/typeobject.c:7261]: > Is it true that this function never sets an exception?

Yes.

> #14836 seems to have an example in which it does set an exception.

No, that's an example where the wrapper descriptor raises an error, not `update_one_slot()`.

Comment (epicfaace) [reply Objects/typeobject.c:7261]: Got it, thanks!

**Review by epicfaace — APPROVED** [review]

**Comment by Yhg1s** [comment]
Thanks, this is a great summary of the code.

**Pull request merged**